### PR TITLE
Reduce number of database queries

### PR DESCRIPTION
**Pull request by lubomir**
These are minor improvements that improved performance while working on the partners ui. I'm not completely sure the approach from the first commit is the best option. Adding `prefetch_related` unconditionally broke the tests.

### CODE REVIEW

File: pdc/apps/release/views.py [hunk 1]
```diff
@@ -748,11 +748,20 @@ class ReleaseVariantViewSet(ChangeSetModelMixin,
...
     deleting a single variant as well as in bulk operations.
     """
-    queryset = models.Variant.objects.all().order_by('id')
+    queryset = models.Variant.objects.order_by('id')
     serializer_class = ReleaseVariantSerializer
     filter_class = filters.ReleaseVariantFilter
     lookup_fields = (('release__release_id', r'[^/]+'), ('variant_uid', r'[^/]+'))
 
+    def get_queryset(self):
+        # Selecting related fields (using a JOIN statement can be done always).
```
Comment (erichuanggit) [inline pdc/apps/release/views.py]: I think this case not only in release variant, why not move to `common`?

**Comment by lubomir** [comment]
I have added a commit which moves part of the code to `StrictQueryParamMixin`. I'm not sure it is a good change, though.

**Comment by ycheng-aa** [comment]
@lubomir Should we continue with pull request?

**Comment by lubomir** [comment]
Let's abandon this. It is not really a problem right now.

**Pull request closed without merging**